tiledb/cloud/file.py

```diff
@@ -36,6 +39,41 @@ def create_file(
         raise tiledb_cloud_error.check_exc(exc) from None
 
 
+def export_file_local(
+    uri: str,
+    output_uri: str,
+    timestamp: Optional[tuple] = None,
+    async_req: bool = False,
+) -> models.FileExported:
+    """
+    Exports a TileDB File back to its original file format
+    :param uri: The ``tiledb://...`` URI of the file to export
+    :param output_uri: output file uri
+    :param tuple timestamp: (default None) If int, open the array at a given TileDB
+        timestamp. If tuple, open at the given start and end TileDB timestamps.
+    :param async_req: return future instead of results for async support
+    :return: FileExported details, including output_uri
+    """
+    try:
+        ctx = client.Ctx()
+        vfs = tiledb.VFS(ctx=ctx)
+        with tiledb.open(uri, ctx=ctx, timestamp=timestamp) as A:
+            file_size = A.meta["file_size"]
+            # Row major partial export of single attribute
+            iterable = A.query(
+                attrs=["contents"], return_incomplete=True, order="C"
+            ).multi_index[0:file_size]
+
+            fh = vfs.open(output_uri, "wb")
+            for part in iterable:
+                vfs.write(fh, part["contents"])
```

Comment:
This will write file contents but what about the extension? Don't we need to take into account the mime type for this?

Reply:
For simplicity I was just forcing the user to provide the full local path